plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncPartsHandler.java

```diff
@@ -56,8 +57,8 @@ public class AsyncPartsHandler {
      * @param inputStreamContainers Checksum containers
      * @param statsMetricPublisher sdk metric publisher
      * @return list of completable futures
-     * @throws IOException thrown in case of an IO error
      */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
     public static List<CompletableFuture<CompletedPart>> uploadParts(
```

Comment:
Do we want to evaluate if we want to use `BufferedInputStream`  for `Normal` priority as well ? It will enable SDK retries at the cost of the increased memory (need to check numbers around same) . Okay taking this as a follow up as well .

Reply:
Yes, this looks like a better thing to do and for sure it is an interesting idea. I had this thought as well earlier but due to some impact even though minimal, I dropped it. But now looking at the complexity of code it looks like we would be better off with BufferedInputStream. Also, with this idea, we would have retries at more granular level i.e on each buffer.

Reply:
I am assuming with controlled uploads the impact of memory overhead due to `BufferedInputStream ` should be limited

Reply:
Yes, absolutely. For a 16 core instance, total number of permits will be 16*4=64 and if lets say all permits are occupied for transfer then will have have (16MB part size + roughly 10% metadata overhead) * 64 MB amount of memory consumed at max.